packages/grid/_modules_/grid/hooks/utils/useCurrentPageRows.ts

```diff
@@ -41,11 +42,19 @@ export const useCurrentPageRows = (
 ) => {
   const response = getCurrentPageRows(apiRef, props);
 
+  const lookup = React.useMemo(() => {
```

Comment:
If you put non-trivial computation here they will be computed for each hook that calls `useCurrentPageRows` and will not take advantage of the selector memoization.
For very large list of rows I think this is to be avoided.

If you are only using it in 1 or 2 `useCurrentPageRows`, I would move the `lookup` computation to the hook / components which calls `useCurrentPageRows`.

If you are using it in more, I would move the `lookup` computation in `getCurrentPageRows` in selectors (one for the paginated version and one for the non-paginated version).